spec-dtslint/util/pipe-spec.ts

```diff
@@ -75,3 +75,15 @@ it('should enforce types for the 8th argument', () => {
 it('should enforce types for the 9th argument', () => {
   const o = pipe(a('0', '1'), a('1', '2'), a('2', '3'), a('3', '4'), a('4', '5'), a('5', '6'), a('6', '7'), a('7', '8'), a('#', '9')); // $ExpectError
 });
+
+it('should return generic OperatorFunction', () => {
+  const customOperator = <T>() => (a: Observable<T>) => a;
+
+  const staticPipe = pipe(customOperator());
+  const o = of('foo').pipe(staticPipe); // $ExpectType Observable<string>
+});
+
+it('should return generic function', () => {
+  const func = pipe((value: string) => value);
+  const value = func("foo"); // ExpectType "foo"
```

Comment:
`$ExpectType`

Reply:
oppps... but, running dtslint does not throw error. I wonder if there is a way to prevent this kind of typo.

Reply:
No error, because it's just seen as a normal comment. At some stage, I'll write a TSLint rule to pick this up, but dealing with comments in the AST is tedious.